android/src/main/java/io/mosip/tuvali/wallet/Wallet.kt

```diff
@@ -1,325 +1,94 @@
-
-  private var central: Central
-
-  //default mtu is 23 bytes and the allowed data bytes is 20 bytes
-  private var maxDataBytes = 20
-  private val mtuValues = arrayOf(512, 185, 100)
-
-  private val retryDiscoverServices = BackOffStrategy(maxRetryLimit = 5)
-
-  private enum class CentralCallbacks {
-    ON_DESTROY_SUCCESS_CALLBACK
-  }
-
-  private val callbacks = mutableMapOf<CentralCallbacks, () -> Unit>()
-
-  init {
-    central = Central(context, this@Wallet)
-    handlerThread.start()
-    transferHandler = TransferHandler(handlerThread.looper, central, Verifier.SERVICE_UUID, this@Wallet)
-  }
-
-  fun stop(onDestroy: () -> Unit) {
-    callbacks[CentralCallbacks.ON_DESTROY_SUCCESS_CALLBACK] = onDestroy
-    central.stop()
-    handlerThread.quitSafely()
-  }
-
-  fun startScanning() {
-    central.scan(Verifier.SERVICE_UUID)
-  }
-
-  fun writeToIdentifyRequest() {
-    val publicKey = walletCryptoBox.publicKey()
-    secretsTranslator = walletCryptoBox.buildSecretsTranslator(verifierPK)
-    val nonce = secretsTranslator?.nonce
-    central.write(Verifier.SERVICE_UUID, GattService.IDENTIFY_REQUEST_CHAR_UUID, nonce!! + publicKey!!)
-    Log.d(logTag, "Started to write - generated nonce ${
-      Hex.toHexString(nonce)
-    }, Public Key of wallet: ${Hex.toHexString(publicKey)}")
-  }
-
-  override fun onScanStartedFailed(errorCode: Int) {
-    Log.d(logTag, "onScanStartedFailed: $errorCode")
-    //TODO: Handle error
-  }
-
-  private val connectionMutex = Object()
 
-  @Volatile
-  private var connectionState = VerifierConnectionState.NOT_CONNECTED
+  override fun startConnection( uri: String) {
+    Log.d(logTag, "startConnection with firstPartOfVerifierPK $uri at ${System.nanoTime()}")
 
-  private enum class VerifierConnectionState {
-    NOT_CONNECTED, CONNECTING, CONNECTED
-  }
+    tryExecuteSync.run {
+      val openId4vpURI = OpenId4vpURI(uri)
 
-  override fun onDeviceFound(device: BluetoothDevice, scanRecord: ScanRecord?) {
-    synchronized(connectionMutex) {
-      if (connectionState != VerifierConnectionState.NOT_CONNECTED) {
-        Log.d(logTag, "Device found is ignored $device")
-        return
+      if(!openId4vpURI.isValid()) {
+        throw InvalidURIException("Received Invalid URI: $uri")
       }
-      val scanResponsePayload = scanRecord?.getServiceData(ParcelUuid(Verifier.SCAN_RESPONSE_SERVICE_UUID))
-      val advertisementPayload = scanRecord?.getServiceData(ParcelUuid(Verifier.SERVICE_UUID))
 
-      if (advertisementPayload != null && isSameAdvPayload(advertisementPayload) && scanResponsePayload != null) {
-        setVerifierPK(advertisementPayload, scanResponsePayload)
-        central.connect(device)
-        connectionState = VerifierConnectionState.CONNECTING
-      } else {
-        Log.d(logTag, "AdvIdentifier($advPayload) is not matching with peripheral device adv")
+      if (bleCommunicator == null) {
```

Comment:
I wonder if we need this check, we can always create WalletBleCommunicator object when startConnection is called. There shouldn't be any case where startConnection will be called and we still had `communicator != null`